app-arch/tar/patches/tar-1.29.patchset

```diff
@@ -5,26 +5,23 @@ Subject: gcc2 patch
 
 
 diff --git a/lib/wordsplit.h b/lib/wordsplit.h
-index d64cf2f..c93f53d 100644
+index e0d2559..acf1f56 100644
 --- a/lib/wordsplit.h
 +++ b/lib/wordsplit.h
-@@ -34,9 +34,9 @@ struct wordsplit
-   const char *ws_escape;
-   void (*ws_alloc_die) (struct wordsplit * wsp);
-   void (*ws_error) (const char *, ...)
--                   __attribute__ ((__format__ (__printf__, 1, 2)));
-+;
-   void (*ws_debug) (const char *, ...)
--                   __attribute__ ((__format__ (__printf__, 1, 2)));
-+;
+@@ -22,7 +22,7 @@
  
-   const char **ws_env;
-   const char *(*ws_getvar) (const char *, size_t, void *);
+ #include <stddef.h>
+ 
+-#if 2 < __GNUC__ + (7 <= __GNUC_MINOR__)
++#if 2 < __GNUC__ + (7 <= __GNUC_MINOR__) - __HAIKU__
```

Comment:
This is a tiny hack. We need the **`#if`** expression to be false for gcc2 and true for gcc4 (or later).
* On x86_gcc2, `__GNU__`=2 and `__GNUC_MINOR__`=95 so without the patch we would be evaluating  (2 < 2 + 1). This is true, but we want it to be false. Hence the - __HAIKU__, which only has an impact on x86_gcc2.
* On x86 and x97_64, i.e. with gcc4, and even with the patch, we evaluate (2 < 4 - 1) or (2 < 4 + 1 - 1), which are both true. If we were using gcc3 on then it would not have been possible to use the hack.

I know, this is a strange hack, but it is not so dirty and it works like a charm when __HAIKU__ is defined ;-)